common/src/main/java/org/opensearch/ml/common/utils/GsonUtil.java

```diff
@@ -0,0 +1,30 @@
+package org.opensearch.ml.common.utils;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.stream.JsonReader;
+
+public class GsonUtil {
```

Comment:
Not quite sure what are we exactly achieving with extending this class? May be we can add more comments to describe importance of this class?

Reply:
I think this is common pattern to create a util class instead of `new` the target class everywhere, the beneficial are:
1. Less memory usage since only have one Gson instance in the GsonUtil.
2. Better encapsulation and better code organization since all Gson related code are in one place.
3. Cleaner code: no need to create Gson instance everywhere.

Reply:
I'm not quite sure about the beneficial pointed out here:

1. We are already doing the same [here](https://github.com/opensearch-project/ml-commons/blob/2.x/common/src/main/java/org/opensearch/ml/common/utils/StringUtils.java#L27-L29). I don't think we improved anything specially here. 
2. In all the `fromJson` methods we are just invoking the `fromJson` method from `Gson`. So I don't think we are achieving better code organization here rather than adding extra code. 
3. Same as 1

Reply:
I missed this one, but I find another case [here](https://github.com/opensearch-project/ml-commons/blob/ff0d73460ee03e7f550c9383529a6627f4ac7ed3/ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java#L55). I think if we all follow the principle to use the Gson in StringUtils, then this GsonUtil is no more needed, I'll change this back but let's pay attention to this in the future code review.